modules/back-end/src/Domain/SemanticPatch/FlagComparer.cs

```diff
@@ -147,13 +161,18 @@ public static IEnumerable<FlagInstruction> CompareFallthrough(Fallthrough origin
     {
         var instructions = new List<FlagInstruction>();
 
-        var isFallThroughChanged =
-            original.DispatchKey != current.DispatchKey ||
-            CompareRolloutVariations(original.Variations, current.Variations);
+        var isFallThroughChanged = CompareRolloutVariations(original.Variations, current.Variations);
 
         if (isFallThroughChanged)
         {
-            instructions.Add(new DefaultVariationInstruction(current));
+            var rolloutVariations = new DefaultRuleRolloutVariations { RolloutVariations = current.Variations };
+            instructions.Add(new UpdateDefaultRuleVariationOrRolloutInstruction(rolloutVariations));
+        }
+
+        // compare dispatch key
+        if (original.DispatchKey != current.DispatchKey && current.DispatchKey != null)
```

Comment:
What if we unset the `DispatchKey`, in this case `current.DispatchKey` will be null ?

Reply:
If you set a rule from percentage rollout to a specific variation, yes, the DispatchKey would be set to null